javasource/queue/helpers/JobToQueueAdder.java

```diff
@@ -108,8 +108,17 @@ public void setTimeZone(IContext context, ILogNode logger) {
 		List<String> timeZoneList = Arrays.asList(TimeZone.getAvailableIDs());
 		boolean timeZoneExists = timeZoneList.stream().anyMatch(tz -> tz.equals(timeZoneID));
 		
+		boolean useDstCorrection = constantsRepository.useDstIfAppliccable();
+		int rawOffset  = TimeZone.getTimeZone(timeZoneID).getRawOffset();
+		int dstCorrection = 0;
+		
+		if (useDstCorrection) {
+			dstCorrection = TimeZone.getTimeZone(timeZoneID).getDSTSavings();
+		}
+		
 		if (timeZoneExists) {
-			context.getSession().setTimeZone(-TimeZone.getTimeZone(timeZoneID).getRawOffset()/1000/60);
+			int offset  = rawOffset + dstCorrection;
```

Comment:
Confusing the `dstCorrection` variable is used here even if not applicable. I would isolate all DST related calculations in the if statement. This can be done by defining the `offset` variable before and add the DST to the offset if applicable. Later, add the timezone to the offset (raw offset).